### ISSUE TITLE
NDEFReader not found in window

### ISSUE THREAD

**Issue by bilafish** [GoogleChrome/samples#677]
Hi, I seem to have some issue detecting the NFC feature support using the code snippet:
`if ('NDEFReader' in window) { /* Scan NFC tags */ }
if ('NDEFWriter' in window) { /* Write NFC tags */ }`

However, when I visit https://googlechrome.github.io/samples/web-nfc/ on the same browser it seems to work.

Any idea what may be the issue here?

**Comment by bilafish** [comment]
I've figured the issue, seems that you need to enable Experimental Web Platform features flag on the Chrome Beta browser as well. Found out from this commit: https://github.com/GoogleChrome/samples/commit/bb1c58ef8ede07e56c8ef075d0b477a917764782

**Comment by beaufortfrancois** [comment]
@bilafish You may want to make sure Chrome Beta is up to date, as you should be able to use Web NFC without the flag. Can you give it a try?

**Comment by bilafish** [comment]
@beaufortfrancois Sure, let me try again later. Will let you know how it goes.

**Comment by bilafish** [comment]
@beaufortfrancois I've just tried again. 
It seems NDEFReader cannot be found when I tried with Experimental Web Platform features flag disabled and webNFC flag enabled. So both `if ('NDEFReader' in window)` or `new NDEFReader()` won't work.

However, when I enabled Experimental Web Platform features flag, scanning seems to work.
I'm using Chrome Beta v81.0.4044.42, there aren't any pending updates on Google Play store.

**Comment by beaufortfrancois** [comment]
Using Chrome Beta v81.0.4044.42, Web NFC is available for me at https://googlechrome.github.io/samples/web-nfc/
Note that it not used on this website, you need either to enable the flag OR [request a token for origin trial](https://web.dev/nfc/#use)

See screenshot below

![image](https://user-images.githubusercontent.com/634478/76218935-53eb8280-6215-11ea-9c16-48b54a3268e5.png)

**Comment by bilafish** [comment]
I see, in that case there could be an issue with my origin trial token since I've already included that in the `<head>` of the page. 
Thanks for your help @beaufortfrancois ! 🙇

**Comment by beaufortfrancois** [comment]
FWIW, the `content` attribute which should contains the token is empty at https://webnfc.now.sh/

![image](https://user-images.githubusercontent.com/634478/76303142-88b51380-62c1-11ea-8a97-eee3fd64747c.png)

**Comment by bilafish** [comment]
Yeah, it seems the .env variable was not injected during the build step of deployment. Have fixed it, now it works without the Experimental Web Platform feature flag enabled. Thanks!

**Comment by beaufortfrancois** [comment]
Woohoo!

**Comment by OlivierGrenoble** [comment]
I have the same issue. I'm using Chrome Beta 81.0.4044.66 and I have enabled the flag #experimental-web-platform-features

I have noticed that Chrome didn,'t ask me the permission to access NFC.

When the line 
  if ('NDEFReader' in window) {
is called, I get "NDEFReader is not defined" 

Shall I try to set the origin trial token?

**Comment by beaufortfrancois** [comment]
According to https://chromiumdash.appspot.com/commit/41de398fe3b045e38fbad40cb2b4b55107764fd0, the flag `#experimental-web-platform-features` flag should be all it's needed to enable Web NFC.

Can you share page / screenshots / console logs?

**Comment by OlivierGrenoble** [comment]
Here is the function called when a button is clicked:

```
function readNfcTagContent() {

  console.log("readNfcTagContent");
	
	if ('NDEFReader' in window) {
		console.log("OK");
	} else {
		console.log("NOK");
	}
	
	const reader = new NDEFReader();  
	...
```

And here is the exception:
script.js:153 Uncaught ReferenceError: NDEFReader is not defined
    at readNfcTagContent (script.js:153)
    at HTMLButtonElement.onclick ((index):21)

The Console shows:
script.js:47 chromeVersion 81
script.js:145 readNfcTagContent
script.js:150 NOK

I have restarted Chrome and rebooted the phone but it didn't help.

**Comment by beaufortfrancois** [comment]
Is your script loaded from https origin or localhost?

**Comment by OlivierGrenoble** [comment]
I'm using localhost (http://localhost:8080/)

**Comment by beaufortfrancois** [comment]
![image](https://user-images.githubusercontent.com/634478/76861063-47da7300-685c-11ea-8bf6-22c299ec23b3.png)

Can you share similar screenshot as mine?
As you can see, it works fine on my side.

**Comment by OlivierGrenoble** [comment]
Here is mine. As you can see it doesn't work on my side :o(

<img width="958" alt="screenshot" src="https://user-images.githubusercontent.com/10893994/76863065-49596a80-685f-11ea-9426-0be03b13218a.png">

**Comment by beaufortfrancois** [comment]
Can you check that the web platform features flag is enabled?

![image](https://user-images.githubusercontent.com/634478/76869795-1d42e700-6869-11ea-8015-0f7f0ce66c48.png)

**Comment by OlivierGrenoble** [comment]
Yes it is enabled:

![Screenshot_20200317-161246](https://user-images.githubusercontent.com/10893994/76870667-4d3eba00-686a-11ea-9413-038cdb074141.png)

**Comment by OlivierGrenoble** [comment]
FYI, in the past, on this phone I have enabled the WebNFC flag. It is now disabled.

**Comment by beaufortfrancois** [comment]
May I ask you to re-install Chrome Beta in case some previous configurations is badly handled on Chrome's side?

**Comment by OlivierGrenoble** [comment]
Hey I have found a workaround! I have clicked on "Reset all to default", relaunched Chrome and then reenabled the flag. Now it works!
There is a bug somewhere in Chrome but it is good to know this workaround!

**Comment by beaufortfrancois** [comment]
Ahhh! I wonder if there were some other flags enabled as well.

**Comment by OlivierGrenoble** [comment]
I don't remember enabling other flags. I'm happy that I can now continue my devs.

**Comment by beaufortfrancois** [comment]
Me too. I'm glad we've finally figured it out ;)

**Comment by OlivierGrenoble** [comment]
Yeah. Thanks for your help and keep safe for the virus.

**Comment by ChrisVonFrosta** [comment]
Hello,
the nfc scan with my mobile phone works on https://googlechrome.github.io/samples/web-nfc/.

I tried to build this on my Websever (http and https). this not working.
Tried it with the origin trial token from this webseit and registered my own one...

On the own hosted site i got the error: "Web NFC is not available. Please make sure the "Experimental Web Platform features" flag is enabled onb Android.

This feature is eabled but on https://googlechrome.github.io/samples/web-nfc/ it works without this flag.

I even tried to reduce the own example to only check for feature abilliy
````
if ('NDEFReader' in window) {
		console.log("OK");
	} else {
		console.log("NOK");
	}
	
	const reader = new NDEFReader(); 
````

**Comment by ChrisVonFrosta** [comment]
> 
> 
> Hello,
> the nfc scan with my mobile phone works on https://googlechrome.github.io/samples/web-nfc/.
> 
> I tried to build this on my Websever (http and https). this not working.
> Tried it with the origin trial token from this webseit and registered my own one...
> 
> On the own hosted site i got the error: "Web NFC is not available. Please make sure the "Experimental Web Platform features" flag is enabled onb Android.
> 
> This feature is eabled but on https://googlechrome.github.io/samples/web-nfc/ it works without this flag.
> 
> I even tried to reduce the own example to only check for feature abilliy
> 
> ```
> if ('NDEFReader' in window) {
> 		console.log("OK");
> 	} else {
> 		console.log("NOK");
> 	}
> 	
> 	const reader = new NDEFReader(); 
> ```

Got it! 
Don't try connect over http - use https!

**Comment by Colonel666** [comment]
in my case It worked only with https
